### PR TITLE
drivers: gpio: nrfx: Use mask to determine if sense is used

### PR DESCRIPTION
**Pull request by nordic-krch**
#31823 added configuration to use sense mechanism for edge interrupts but solution that was used back then was for all or none. It was already raised back then that per-pin solution is needed (https://github.com/zephyrproject-rtos/zephyr/pull/31823#pullrequestreview-608657733).

This PR do this by adding device tree mask which can be used to configure which pins will use sensing for edge interrupts. Added test configuration to validate that.

Maybe there is a way to have it more user friendly by allowing per-pin configuration in device tree instead of mask. Any device tree experts suggestions are welcomed.

### CODE REVIEW

**Review by mniestroj — APPROVED** [review]
LGTM! Tested as well.

File: drivers/gpio/gpio_nrfx.c [hunk 1]
```diff
@@ -370,7 +374,8 @@ static const struct gpio_driver_api gpio_nrfx_drv_api_funcs = {
 			GPIO_PORT_PIN_MASK_FROM_DT_INST(id),		\
 		},							\
 		.port = (NRF_GPIO_Type *)DT_INST_REG_ADDR(id),		\
-		.port_num = DT_INST_PROP(id, port)			\
+		.port_num = DT_INST_PROP(id, port),			\
+		.edge_sense = GPIO_EDGE_SENSE(id)			\
```
Comment (anangl) [inline drivers/gpio/gpio_nrfx.c:378]: How about:
```suggestion
		.edge_sense = DT_INST_PROP_OR(id, sense_edge_mask, 0)	\
```

File: dts/bindings/gpio/nordic,nrf-gpio.yaml [hunk 2]
```diff
@@ -17,6 +17,13 @@ properties:
     "#gpio-cells":
       const: 2
 
+    sense-edge-mask:
+      required: false
+      type: int
+      description: |
+        Mask of pins that use GPIO sense for edge detection. Pins not included
+        in the mask use GPIOTE event.
```
Comment (anangl) [inline dts/bindings/gpio/nordic,nrf-gpio.yaml:25]:
```suggestion
        Mask of pins that use the GPIO sense mechanism for edge detection.
        Pins not included in the mask use GPIOTE channels in event mode.
```

File: drivers/gpio/gpio_nrfx.c (hunk 1 above)
Comment (nordic-krch) [reply drivers/gpio/gpio_nrfx.c]: done

File: dts/bindings/gpio/nordic,nrf-gpio.yaml (hunk 2 above)
Comment (nordic-krch) [reply dts/bindings/gpio/nordic,nrf-gpio.yaml]: done

**Review by anangl — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**